app/controllers/merchant/items_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Merchant::ItemsController < Merchant::BaseController
+
+  def index
+    @merchant = Merchant.find(current_user[:merchant_id])
+  end
+
+  def new
+    @merchant = Merchant.find(current_user[:merchant_id])
+  end
+
+  def add_item
+    @merchant = Merchant.find(current_user[:merchant_id])
+    item = @merchant.items.new(item_params)
+    if item.save
+      flash[:notice] = "#{item.name} has been added to your catalog."
+      redirect_to "/merchant/items"
+    else
+      flash[:notice] = "Unable to add item: #{item.errors.full_messages.to_sentence}."
```

Comment:
If we want, we can convert all our error messages to use this `.errors.full_messages.to_sentence` since it returns specific information like `name can't be blank`